wallet/bip32_test.go

```diff
@@ -0,0 +1,108 @@
+package wallet
+
+import (
+	"crypto/ed25519"
+	"crypto/sha512"
+	"encoding/hex"
+	"github.com/spacemeshos/smkeys/bip32"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/xdg-go/pbkdf2"
+)
+
+var goodSeed = []byte("abandon abandon abandon abandon abandon abandon abandon abandon ")
+
+func generateTestMasterKeyPair() (*EDKeyPair, error) {
+	mnemonic := "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"
+	password := "Winning together!"
+	seed := pbkdf2.Key([]byte(mnemonic), []byte("mnemonic"+password), 2048, 64, sha512.New)
+	return NewMasterKeyPair(seed)
+}
+
+func TestNewMasterBIP32EDKeyPair(t *testing.T) {
+	// first iteration
+	keyPair1, err := generateTestMasterKeyPair()
+	require.NoError(t, err)
+	require.NotEmpty(t, keyPair1)
+
+	// second iteration
+	keyPair2, err := generateTestMasterKeyPair()
+	require.NoError(t, err)
+	require.NotEmpty(t, keyPair2)
+
+	msg := []byte("master test")
+
+	// Testing the private key signature generated from the first iteration and verifying with public key from the second iteration
+	sig1 := ed25519.Sign(ed25519.PrivateKey(keyPair1.Private), msg)
+	valid1 := ed25519.Verify(ed25519.PublicKey(keyPair2.Public), msg, sig1)
+	require.True(t, valid1)
+
+	// Same test with swapped private and public key
+	sig2 := ed25519.Sign(ed25519.PrivateKey(keyPair2.Private), msg)
+	valid2 := ed25519.Verify(ed25519.PublicKey(keyPair1.Public), msg, sig2)
+	require.True(t, valid2)
+}
+
+func TestNonHardenedPath(t *testing.T) {
+	path1Str := "m/44'/540'/0'/0'/0'"
+	path1Hd, err := StringToHDPath(path1Str)
+	require.NoError(t, err)
+	require.True(t, IsPathCompletelyHardened(path1Hd))
+
+	path2Str := "m/44'/540'/0'/0'/0"
+	path2Hd, err := StringToHDPath(path2Str)
+	require.NoError(t, err)
+	require.False(t, IsPathCompletelyHardened(path2Hd))
+}
+
+// Test that path string produces expected path and vice-versa
+func TestPath(t *testing.T) {
+	path1Str := "m/44'/540'/0'/0'/0'"
+	path1Hd, err := StringToHDPath(path1Str)
+	require.NoError(t, err)
+	path := DefaultPath()
+	path2Hd := path.Extend(BIP44HardenedAccountIndex(0))
```

Comment:
Would it be OK to use a "fluent" call here? (same in the next test)

```suggestion
	path2Hd := DefaultPath().Extend(BIP44HardenedAccountIndex(0))
```

Reply:
doesn't work because the method expects a pointer receiver